### PR TITLE
Functionality for resetting the simulation phase

### PR DESCRIPTION
**Pull request by joelkuiper**
(work in progress for story #199)

### CODE REVIEW

**Comment by joelkuiper** [comment]
This misses some event classes, will push tomorrow morning!

File: mtc-mcmc/src/main/java/org/drugis/mtc/yadas/YadasModel.java [hunk 1]
```diff
@@ -39,14 +39,18 @@
 import java.util.Map;
 
 import org.apache.commons.math3.random.JDKRandomGenerator;
+import org.drugis.common.threading.AbortedException;
 import org.drugis.common.threading.AbstractIterativeComputation;
 import org.drugis.common.threading.IterativeTask;
 import org.drugis.common.threading.NullTask;
+import org.drugis.common.threading.RestartableIterativeTask;
```
Comment (gertvv) [inline mtc-mcmc/src/main/java/org/drugis/mtc/yadas/YadasModel.java]: I was missing this from the drugis-common pull request but I see that a commit just appeared.

Comment (joelkuiper) [inline mtc-mcmc/src/main/java/org/drugis/mtc/yadas/YadasModel.java]: Yep, just pushed!

On Tue, Apr 24, 2012 at 10:32 AM, Gert van Valkenhoef <
reply@reply.github.com

File: mtc-mcmc/src/main/java/org/drugis/mtc/yadas/YadasModel.java [hunk 2]
```diff
@@ -131,13 +138,25 @@ public BurnInTask(int chain) {
...
+	private class SimulationTask extends RestartableIterativeTask {
 		public SimulationTask(int chain) {
 			super(new SimulationChain(chain), "simulation:" + chain);
 			setReportingInterval(d_reportingInterval);
 		}
 	}
+	
+	private class SimpleRestartableSuspendableTask extends SimpleSuspendableTask {
 
+		public SimpleRestartableSuspendableTask(Runnable runnable, String string) {
```
Comment (gertvv) [inline mtc-mcmc/src/main/java/org/drugis/mtc/yadas/YadasModel.java:179]: The IterativeSuspendable does quite a bit of work to be able to suspend the computation. I was worried that this does not carry over to this class, but it seems that you use RestartableIterativeTask for the actual iterative computation instead. Does that handle suspend correctly?

**Comment by gertvv** [comment]
Looks good in general, just take note of my remarks on drugis-common about the interfaces, and make sure that suspending the computation still works!

**Comment by joelkuiper** [comment]
Is there a simple way of trying the suspending? Right now
RestartableIterativeTask simply inherits from IterativeTask

On Tue, Apr 24, 2012 at 10:41 AM, Gert van Valkenhoef <
reply@reply.github.com

**Comment by gertvv** [comment]
To test the suspend, start an MTC analysis, this one will use all your available cores (if you have <= 4). Now, if you start a second one, that one should start immediately, and the first one should be suspended (and not progress until the second one has finished).

**Pull request merged**